controllers/disruption_controller.go

```diff
@@ -462,6 +468,7 @@ func (r *DisruptionReconciler) createChaosPods(instance *chaosv1beta1.Disruption
 		// get IDs of targeted containers or all containers
 		targetContainers, err = utils.GetTargetedContainersInfo(&pod, instance.Spec.Containers)
 		if err != nil {
+			r.log.Debugw("unable to get target pod container ID", "targetPodStatus", pod.Status, "targetPodName", target, "targetPodNamespace", instance.Namespace)
```

Comment:
The error will get logged at error in the top-level reconcile loop, but I don't want to stuff all this pod info into returns so that we can log it up there, so I'm debug logging it here. reasonable?

Reply:
Shouldn't this be part of the error informations instead?
Could we create a custom error type to encapsulate error and provide contextual informations that will be logged as expected?

It feels we are always logging + erroring, that is a code smell we are missing something from my POV.

The error itself should host all informations expected to troubleshoot, or a log should be here and no error returned.

Here we are always doing both, it feels weird from my POV.

Reply:
If there's a nice way to stuff the tags we want into the error, so that it can be logged at the top level, that would look cleanest, I agree.

Would that be alright in a follow-up? I primarily want to get the node failure change merged, this logging was all just in pursuit of debugging that

Reply:
What I was thinking about was creating a custom error type with proper contextual properties or random `map[string]string` contextual fields.